dev/tasks/python-wheels/github.osx.yml

```
@@ -31,88 +43,107 @@ jobs:
       - name: Checkout Arrow
         run: |
           set -ex
-
           git clone --no-checkout {{ arrow.remote }} arrow
           git -C arrow fetch -t {{ arrow.remote }} {{ arrow.branch }}
           git -C arrow checkout FETCH_HEAD
           git -C arrow submodule update --init --recursive
 
-      - name: Build wheel
+      - name: Install System Dependencies
+        run: brew install bison ninja
+
+      # Restore from cache the previously built ports.
+      # If cache-miss, download and build vcpkg (aka "bootstrap vcpkg").
+      - name: Restore from Cache and Install Vcpkg
+        # Download and build vcpkg, without installing any port.
+        # If content is cached already, it is a no-op.
+        uses: kszucs/run-vcpkg@main
+        with:
+          # Required to prevent cache eviction on crossbow's main branch
+          # where we build pre-build the vcpkg packages
+          setupOnly: true
+          doNotSaveCache: true
+          appendedCacheKey: "-macos-{{ macos_deployment_target }}"
+          vcpkgDirectory: {{ "${{ github.workspace }}/vcpkg" }}
+          vcpkgGitCommitId: fced4bef1606260f110d74de1ae1975c2b9ac549
+
+      - name: Patch Vcpkg Ports
         run: |
-          set -ex
-
-          git clone https://github.com/matthew-brett/multibuild
-          git -C multibuild checkout 8882150df6529658700b66bec124dfb77eefca26
-
-          # source utilities required for wheel builds
-          export CONFIG_PATH=`pwd`/arrow/dev/tasks/python-wheels/osx-build.sh
-          source multibuild/common_utils.sh
-          source multibuild/travis_osx_steps.sh
-
-          # provided by travis_osx_steps
-          before_install
-
-          brew install \
-            automake \
-            bison \
-            boost \
-            ccache \
-            cmake \
+          cd $VCPKG_ROOT
+          if ! git apply --reverse --check --ignore-whitespace ../arrow/ci/vcpkg/ports.patch; then
+            git apply --ignore-whitespace ../arrow/ci/vcpkg/ports.patch
```

This depends on `VCPKG_ROOT` path. How about the following?

```suggestion
          vcpkg_patch_file=$PWD/arrow/ci/vcpkg/ports.patch
          cd $VCPKG_ROOT
          if ! git apply --reverse --check --ignore-whitespace ${vcpkg_patch_file}; then
            git apply --ignore-whitespace ${vcpkg_patch_file}
```

---

LGTM